src/main/resources/assets/js/search.js

```diff
@@ -0,0 +1,57 @@
+window.onpageshow = function() {
+  const el = document.getElementById('search');
+  el.focus();
+
+  const params = new URLSearchParams(window.location.search);
+  const terms = params.get('q');
+
+  if (terms) {
+    el.value = terms;
+    search(terms);
+  }
+};
+
+function search(terms) {
+  const results = idx.search(terms).map(item => {
+    const document = documents.find(post => item.ref === post.href);
+    return { score: item.score, href: document.href, type: document.type, title: document.title };
+  });
+
+  const div = document.getElementById('search-results');
+
+  if (results.length === 0) {
+    div.appendChild(createNoResultsElement());
+  } else {
+    results.forEach(result => {
+      div.appendChild(createResultElement(result));
+    });
+  }
+}
+
+function createResultElement(result) {
```

Comment:
I'm very open for a better presentation ;)
We might e.g. also present the score in forms of stars (I haven't seen a score higher that 3.3, so having a range from zero to 5 would work). Note that the results are already sorted on score.

Reply:
I'm not sure whether or not making the search score visible would be beneficial. I'd rather have some kind of preview, like Google has, that would make it easier for a user to determine if a particular search result is actually what they're looking for. But I don't think that is necessary for now. I think this is already a valuable addition.